h5pt.go

```diff
@@ -118,11 +119,27 @@ func (t *Table) ReadPackets(start, nrecords int, data interface{}) error {
 // Struct values must only have exported fields, otherwise extractValues will
 // panic.
 func extractValues(ad *appendData, data interface{}) error {
+	if data, ok := data.(interface{ CMarshal() ([]byte, error) }); ok {
+		raw, err := data.CMarshal()
+		if err != nil {
+			return err
+		}
+
+		tmp := make([]byte, ad.offset+len(raw))
+		copy(tmp, ad.cBuffer)
+		tmp = ad.cBuffer
```

Comment:
That is actually the wrong way around. I have no idea, why the test cases passed 😆 
Should obviously be `ad.cBuffer = tmp`. But with the proposed change, this will fall away.

Reply:
probably b/c nobody's implementing this made-up interface :)

Reply:
Haha thought this is the coder further below xD Just making noobish comments one after another. Should just shut up xD